### ISSUE TITLE
Xamarin.Mac support question

### ISSUE THREAD

**Issue by RocketPop** [akavache/Akavache#215]
Hi - I'm trying to use akavache on Xamarin.Mac and having no success.
I have tried making a blank Xamarin.Mac "classic api" project and added the one-line ApplicationName setup to the main.cs and put a simple one-line BlobCache.LocalMachine.InsertObject("key", "value"); in the window init.

I get this exception:
```
Unhandled Exception:
System.TypeInitializationException: An exception was thrown by the type initializer for NativeMethods ---> System.Exception: sqlite3.dll was not loaded.
  at SQLitePCL.SQLite3Provider+NativeMethods..cctor () [0x00000] in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at SQLitePCL.SQLite3Provider.SQLitePCL.ISQLite3Provider.sqlite3_open_v2 (System.String filename, System.IntPtr& db, Int32 flags, System.String vfs) [0x00000] in <filename unknown>:0 
  at SQLitePCL.raw.sqlite3_open_v2 (System.String filename, SQLitePCL.sqlite3& db, Int32 flags, System.String vfs) [0x00000] in <filename unknown>:0 
  at Akavache.Sqlite3.Internal.SQLite3.Open (System.String filename, SQLitePCL.sqlite3& db, Int32 flags, IntPtr zVfs) [0x00000] in <filename unknown>:0 
  at Akavache.Sqlite3.Internal.SQLiteConnection..ctor (System.String databasePath, SQLiteOpenFlags openFlags, Boolean storeDateTimeAsTicks) [0x00000] in <filename unknown>:0 
  at Akavache.Sqlite3.Internal.SQLiteConnection..ctor (System.String databasePath, Boolean storeDateTimeAsTicks) [0x00000] in <filename unknown>:0 
  at Akavache.Sqlite3.SQLitePersistentBlobCache..ctor (System.String databaseFile, IScheduler scheduler) [0x00000] in <filename unknown>:0 
  at Akavache.Sqlite3.Registrations+<>c__DisplayClass6.<Register>b__0 () [0x00000] in <filename unknown>:0 
  at System.Lazy`1[Akavache.IBlobCache].InitValue () [0x0015f] in /private/tmp/source-mono-mac-3.12.0-branch-32/bockbuild-mono-3.12.0-branch/profiles/mono-mac-xamarin/build-root/mono-3.12.1/mcs/class/corlib/System/Lazy.cs:164 
[ERROR] FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: An exception was thrown by the type initializer for NativeMethods ---> System.Exception: sqlite3.dll was not loaded.
  at SQLitePCL.SQLite3Provider+NativeMethods..cctor () [0x00000] in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at SQLitePCL.SQLite3Provider.SQLitePCL.ISQLite3Provider.sqlite3_open_v2 (System.String filename, System.IntPtr& db, Int32 flags, System.String vfs) [0x00000] in <filename unknown>:0 
  at SQLitePCL.raw.sqlite3_open_v2 (System.String filename, SQLitePCL.sqlite3& db, Int32 flags, System.String vfs) [0x00000] in <filename unknown>:0 
  at Akavache.Sqlite3.Internal.SQLite3.Open (System.String filename, SQLitePCL.sqlite3& db, Int32 flags, IntPtr zVfs) [0x00000] in <filename unknown>:0 
  at Akavache.Sqlite3.Internal.SQLiteConnection..ctor (System.String databasePath, SQLiteOpenFlags openFlags, Boolean storeDateTimeAsTicks) [0x00000] in <filename unknown>:0 
  at Akavache.Sqlite3.Internal.SQLiteConnection..ctor (System.String databasePath, Boolean storeDateTimeAsTicks) [0x00000] in <filename unknown>:0 
  at Akavache.Sqlite3.SQLitePersistentBlobCache..ctor (System.String databaseFile, IScheduler scheduler) [0x00000] in <filename unknown>:0 
  at Akavache.Sqlite3.Registrations+<>c__DisplayClass6.<Register>b__0 () [0x00000] in <filename unknown>:0 
  at System.Lazy`1[Akavache.IBlobCache].InitValue () [0x0015f] in /private/tmp/source-mono-mac-3.12.0-branch-32/bockbuild-mono-3.12.0-branch/profiles/mono-mac-xamarin/build-root/mono-3.12.1/mcs/class/corlib/System/Lazy.cs:164 
```

I have added Akavache, Akavache.sqlite3 and the various Rx packages via nuget. Am I missing a mac-specific setup step?

ninja edit: I also tried a unified project and it gives an error about the dummy dll is loaded, not the platform specific dll. But I see 4.1.0 is supposed to support unified.. Let me try again and report back.

**Comment by paulcbetts** [comment]
@RocketPop What happens if you create a Unified one? We can't load sqlite3.dylib for some reason. You might need a special .config file to map "sqlite3.dll" => "sqlite3.dylib"

**Comment by RocketPop** [comment]
The unified API test (same steps as above, but moved the insert a bit) results in this exception:
```
Unhandled Exception:
System.Exception: Something went wrong in the build configuration.  This is the bait assembly, which is for referencing by portable libraries, and should never end up part of the app.  Reference the appropriate platform assembly instead.
  at SQLitePCL.SQLite3Provider.SQLitePCL.ISQLite3Provider.sqlite3_open_v2 (System.String filename, System.IntPtr& db, Int32 flags, System.String vfs) [0x00000] in <filename unknown>:0 
  at SQLitePCL.raw.sqlite3_open_v2 (System.String filename, SQLitePCL.sqlite3& db, Int32 flags, System.String vfs) [0x00000] in <filename unknown>:0 
  at Akavache.Sqlite3.Internal.SQLite3.Open (System.String filename, SQLitePCL.sqlite3& db, Int32 flags, IntPtr zVfs) [0x00000] in <filename unknown>:0 
  at Akavache.Sqlite3.Internal.SQLiteConnection..ctor (System.String databasePath, SQLiteOpenFlags openFlags, Boolean storeDateTimeAsTicks) [0x00000] in <filename unknown>:0 
  at Akavache.Sqlite3.Internal.SQLiteConnection..ctor (System.String databasePath, Boolean storeDateTimeAsTicks) [0x00000] in <filename unknown>:0 
  at Akavache.Sqlite3.SQLitePersistentBlobCache..ctor (System.String databaseFile, IScheduler scheduler) [0x00000] in <filename unknown>:0 
  at Akavache.Sqlite3.Registrations+<>c__DisplayClass6.<Register>b__0 () [0x00000] in <filename unknown>:0 
  at System.Lazy`1[Akavache.IBlobCache].InitValue () [0x0015f] in /Users/builder/data/lanes/1248/f2847d5f/source/xamcore/mono/mcs/class/corlib/System/Lazy.cs:164 
[ERROR] FATAL UNHANDLED EXCEPTION: System.Exception: Something went wrong in the build configuration.  This is the bait assembly, which is for referencing by portable libraries, and should never end up part of the app.  Reference the appropriate platform assembly instead.
  at SQLitePCL.SQLite3Provider.SQLitePCL.ISQLite3Provider.sqlite3_open_v2 (System.String filename, System.IntPtr& db, Int32 flags, System.String vfs) [0x00000] in <filename unknown>:0 
  at SQLitePCL.raw.sqlite3_open_v2 (System.String filename, SQLitePCL.sqlite3& db, Int32 flags, System.String vfs) [0x00000] in <filename unknown>:0 
  at Akavache.Sqlite3.Internal.SQLite3.Open (System.String filename, SQLitePCL.sqlite3& db, Int32 flags, IntPtr zVfs) [0x00000] in <filename unknown>:0 
  at Akavache.Sqlite3.Internal.SQLiteConnection..ctor (System.String databasePath, SQLiteOpenFlags openFlags, Boolean storeDateTimeAsTicks) [0x00000] in <filename unknown>:0 
  at Akavache.Sqlite3.Internal.SQLiteConnection..ctor (System.String databasePath, Boolean storeDateTimeAsTicks) [0x00000] in <filename unknown>:0 
  at Akavache.Sqlite3.SQLitePersistentBlobCache..ctor (System.String databaseFile, IScheduler scheduler) [0x00000] in <filename unknown>:0 
  at Akavache.Sqlite3.Registrations+<>c__DisplayClass6.<Register>b__0 () [0x00000] in <filename unknown>:0 
  at System.Lazy`1[Akavache.IBlobCache].InitValue () [0x0015f] in /Users/builder/data/lanes/1248/f2847d5f/source/xamcore/mono/mcs/class/corlib/System/Lazy.cs:164 
```

**Comment by paulcbetts** [comment]
@RocketPop NuGet is doing something dumb, reference the  Xamarin.iOS version

**Comment by RocketPop** [comment]
If I only had a dime for every weird nuget bug I have bumped into :) My project as created by nuget had this reference:
```
    <Reference Include="SQLitePCL.raw">
      <HintPath>packages\SQLitePCL.raw_basic.0.7.1\lib\portable-net45+netcore45+wp8+MonoAndroid10+MonoTouch10+Xamarin.iOS10\SQLitePCL.raw.dll</HintPath>
    </Reference>
```
After manually changing the reference to just be "Xamarin.iOS10" the project runs as expected. I'll verify it worked, but at least it no longer throws. Thanks for the help!

**Comment by paulcbetts** [comment]
@RocketPop Well in this case I don't think it's actually a NuGet "bug" so much as, SQLitePCL.Raw probably doesn't ship a Xamarin.Mac profile, so we're haxing it a bit

**Comment by RocketPop** [comment]
I mis-spoke - more of a nuget quirk :)
So now it does init properly, and I can call InsertObject(), but when awaiting any of the IObservable results such as DownloadUrl() - they never complete. Do I have to reference any other iOS dlls to make that RX plumbing work?

**Comment by RocketPop** [comment]
Guess I'll table this until the level of mac support is more clear :)
